crates/sui-adapter/src/adapter.rs

```diff
@@ -480,294 +504,156 @@ fn process_successful_execution<
-                            // Here it is set to `v+1` and will be incremented in `delete_object`
-                            state_view.log_event(Event::delete_object(
-                                module_id.address(),
-                                module_id.name(),
-                                ctx.sender(),
-                                *obj_id,
-                            ));
-                            match state_view.get_latest_parent_entry_ref(*obj_id) {
-                                Ok(Some((_, previous_version, _))) => {
-                                    changes.insert(
-                                        *obj_id,
-                                        ObjectChange::Delete(
-                                            previous_version,
-                                            DeleteKind::UnwrapThenDelete,
-                                        ),
-                                    );
-                                }
-                                // if the object is not in parent sync, it was wrapped before
-                                // ever being stored into storage. Thus we don't need to mark it
-                                // as being deleted
-                                Ok(None) => (),
-                                // TODO this error is (hopefully) transient and should not be
-                                // a normal execution error
-                                Err(_) => {
-                                    return Err(ExecutionError::new_with_source(
-                                        ExecutionErrorKind::InvariantViolation,
-                                        missing_unwrapped_msg(obj_id),
-                                    ));
-                                }
-                            };
-                        }
-                    }
+            // When an object was wrapped at version `v`, we added an record into `parent_sync`
+            // with version `v+1` along with OBJECT_DIGEST_WRAPPED. Now when the object is unwrapped,
+            // it will also have version `v+1`, leading to a violation of the invariant that any
+            // object_id and version pair must be unique. We use the version from parent_sync and
+            // increment it (below), so we will have `(v+1)+1`, thus preserving the uniqueness
+            (None, WriteKind::Unwrap) => match state_view.get_latest_parent_entry_ref(id) {
+                Ok(Some((_, last_version, _))) => (WriteKind::Unwrap, last_version),
+                // if the object is not in parent sync, it was wrapped before ever being stored into
+                // storage.
+                // we set is_unwrapped to false since the object has never been in storage
+                // and essentially is being created. Similarly, we add it to the newly_generated_ids
+                // set
+                Ok(None) => (WriteKind::Create, SequenceNumber::new()),
```

Comment:
An object gets created and wrapped in a single transaction. But then unwrapped and transferred in another. 
As such, we are in this case because the created+wrapped means no old object and no entry in parent sync. 

This case will go away with lamport time stamps, but we will lose some information as it will swap from `Create` to `Mutate` as we are no longer querying the parent sync. If we want to switch it back to a `Create` we will need to query the locking service (as the parent sync is not really idempotent)